flyflix.py

```diff
@@ -63,11 +65,41 @@ def read_metadata():
         try:
             with metadata_lock:
                 metadata = yaml.safe_load(stream)
+                metadata_as_string()
```

Comment:
Try to keep the locks as minimal as possible. A [`lock` (or mutex)](https://docs.python.org/3/library/threading.html#lock-objects)  should be as granular as possible. I introduced the `metadata_lock` to ensure that only one flask thread modifies the global `metadata` variable at any time. The whole `metadata_as_string()` function can take some time and would block other threads from writing to the variable. Can you make the use of `metadata_lock` more granular?